src/gwc/src/test/java/org/geoserver/gwc/GWCTest.java

```diff
@@ -1195,6 +1203,43 @@ private void assertDispatchMismatch(GetMapRequest request, String expectedReason
                 target.toString().contains(expectedReason));
     }
 
+    /**
+     * Confirms that GWC handling of WMS requests does not enter recursion when encountering a seed
+     * request
+     */
+    @Test
+    public void testGetMapRequest() throws Throwable {
+        GetMapRequest request = new GetMapRequest();
+
+        @SuppressWarnings("unchecked")
+        Map<String, String> rawKvp = new CaseInsensitiveMap(new HashMap<String, String>());
+        rawKvp.put(GeoServerTileLayer.GWC_SEED_INTERCEPT_TOKEN, "true");
+        request.setRawKvp(rawKvp);
+        request.setTiled(true);
```

Comment:
The bug report was actually about not having the tiled request parameter required, e.g.:

```
mediator.getConfig().setRequireTiledParameter(false);
```

Reply:
Fixed